tests/test_lark_lark.py

```diff
@@ -0,0 +1,211 @@
+from __future__ import absolute_import
+
+import os
+from unittest import TestCase, main
+
+from lark import lark, Lark, UnexpectedToken
+from lark.load_grammar import GrammarError
+from lark.lark_validator_visitor import LarkValidatorVisitor
+
+
+# Test that certain previous differences between load_grammar.py and
+# grammars/lark.lark have been resolved.
+class TestLarkLark(TestCase):
+    def setUp(self):
+        lark_path = os.path.join(os.path.dirname(lark.__file__), 'grammars/lark.lark')
+        self.lark_parser = Lark.open(lark_path, parser="lalr")
+
+    def test_01_no_alias_in_terminal_lg(self):
+        g = """start: TERM
+            TERM: "a" -> alias
+            """
+        self.assertRaisesRegex( GrammarError, "Aliasing not allowed in terminals", Lark, g)
+
+    def test_01_no_alias_in_terminal_ll(self):
+        # lark.lark allows aliases in terminals, and rejects them if you run the LarkValidatorVisitor.
+        g = """start: TERM
+            TERM: "a" -> alias
+            """
+        self.lark_parser.parse(g)
+
+    def test_02_no_rule_aliases_below_top_level_lg(self):
+        g = """start: rule
+            rule: ("a" -> alias
+                 | "b")
+            """
+        self.assertRaisesRegex( GrammarError, "Rule 'alias' used but not defined", Lark, g)
+
+    def test_02_no_rule_aliases_below_top_level_ll(self):
+        # lark.lark allows aliases below top-level, and rejects them if you run the LarkValidatorVisitor.
+        g = """start: rule
+            rule: ("a" -> alias
+                 | "b")
+            """
+        self.lark_parser.parse(g)
+
+    def test_04_extend_rule_lg(self):
+        g = """
+            %import .grammars.ab (startab, A, B, expr)
+
+            %extend expr: B A
+        """
+        Lark(g, start='startab', source_path=__file__)
+
+    def test_04_extend_rule_ll(self):
+        g = """
+            %import .grammars.ab (startab, A, B, expr)
+
+            %extend expr: B A
+        """
+        self.lark_parser.parse(g)
+
+    def test_05_extend_term_lg(self):
+        g = """
+            %import .grammars.ab (startab, A, B, expr)
+
+            %extend A: "c"
+        """
+        Lark(g, start='startab', source_path=__file__)
+
+    def test_05_extend_term_ll(self):
+        g = """
+            %import .grammars.ab (startab, A, B, expr)
+
+            %extend A: "c"
+        """
+        self.lark_parser.parse(g)
+
+    def test_06_no_term_templates_lg(self):
+        g = """start: TERM
+        separated{x, sep}: x (sep x)*
+        TERM: separated{"A", " "}
+        """
+        self.assertRaisesRegex( AssertionError, "Tree.'template_usage', .NonTerminal.'separated'.", Lark, g)
```

Comment:
`AssertionErrors` are not something we want to have in the public interface, this should be changed to a `GrammarError`. (although you don't need to do it now in this PR)

Reply:
Fixed in 0fb28f9ad16b461821227f2dbfa71a01d84cf363.  I already had an error for this in `LarkValidatorVisitor`: "_Templates not allowed in terminals_".  Thanks to your hint elsewhere about rules not being inlined because they have multiple children, I realized `template_usage` can never be inlined - the minimum is two children: a `RULE` and a `value`.  Which made it easy to add `TerminalTreeToPattern.template_usage()` to catch the bad use.